tutorial/managing-software.rst

```diff
@@ -0,0 +1,1611 @@
+package information. It's installed by default on Ubuntu systems so we can use
+it directly:
+
+.. code-block:: bash
+
+    dpkg --listfiles ssl-cert
+
+This gives us the following list of files and their directory structure (the
+end of the list is truncated for brevity):
+
+.. code-block:: text
+
+    /.
+    /etc
+    /etc/ssl
+    /etc/ssl/certs
+    /etc/ssl/private
+    /lib
+    diverted by base-files to: /lib.usr-is-merged
+    /lib/systemd
+    /lib/systemd/system
+    /lib/systemd/system/ssl-cert.service
+    /usr
+    /usr/sbin
+    /usr/sbin/make-ssl-cert
+    /usr/share
+    /usr/share/doc
+    /usr/share/doc/ssl-cert
+    /usr/share/doc/ssl-cert/README
+    [...]
+
+If we find a file but we're not sure what package it comes from, ``dpkg`` can
+help us there too! Let's use the example of one of the files from the previous
+output: ``/usr/share/ssl-cert/ssleay.cnf`` and do a search for it using
+``dpkg``:
+
+.. code-block:: bash
+
+    dpkg --search /usr/share/ssl-cert/ssleay.cnf
+
+This will provide us with the package name for the given file:
+
+.. code-block:: bash
+
+    ssl-cert: /usr/share/ssl-cert/ssleay.cnf
+
+Although this seems obvious to us, because we already know the source of this
+file, the ``dpkg`` search function is really useful for tracking down the
+sources of files we don't know about!
+
+Conffiles
+---------
+
+Most of a package's configuration is handled through
+`configuration files <https://www.debian.org/doc/debian-policy/ap-pkg-conffiles.html#automatic-handling-of-configuration-files-by-dpkg>`_
+(often known as **conffiles**). Conffiles often contain things like file paths,
+logs and debugging configuration, kernel parameters (which can be changed to
+optimise system performance), access control, and other configuration settings.
+The actual parameters available will vary from one package to another.
+
+Package conffiles are different from all other files delivered in a package.
+A package may have any number of conffiles (including none!). Conffiles are
+explicitly marked by the package maintainer during development to protect them
+from being automatically upgraded, whilst all other types of files are not --
+any changes you make to regular files *will be overwritten* during an upgrade. 
+
+How upgrades are handled
+------------------------
+
+Since a conffile can be changed by us, we might end up with conflicts when the
+package maintainer changes those same files. Therefore, it's important to
+understand how such conflicts are handled.
+
+We can show the four possible upgrade scenarios using the following table. What
+happens during an upgrade depends on whether the conffile on our system has
+been changed by us ("changed/not changed by user"), and whether the default
+content has been changed by the package maintainer ("same as/changed from
+default"):
+
+========================== =================== =========================
+The conffile is...         **same as default** **changed from default**
```

Comment:
That's a great point. I picked "default" because it made the table more compact but I see the ambiguity I've created by doing that. I've changed it to "changed/not changed by maintainer" -- hopefully this is better!